lib/matplotlib/backend_tools.py

```diff
@@ -273,24 +273,15 @@ def set_figure(self, figure):
             self._id_drag = self.canvas.mpl_connect(
                 'motion_notify_event', self._set_cursor_cbk)
 
-    def _tool_trigger_cbk(self, event):
-        if event.tool.toggled:
-            self._current_tool = event.tool
-        else:
-            self._current_tool = None
-        self._set_cursor_cbk(event.canvasevent)
-
-    def _add_tool(self, tool):
-        """Set the cursor when the tool is triggered."""
-        if getattr(tool, 'cursor', None) is not None:
-            self.toolmanager.toolmanager_connect('tool_trigger_%s' % tool.name,
-                                                 self._tool_trigger_cbk)
-
     def _add_tool_cbk(self, event):
         """Process every newly added tool."""
-        if event.tool is self:
-            return
```

Comment:
Do we not need this check any more? I didn't see any explicit mention of it.

Reply:
ToolSetCursor.cursor is None, so the new condition (`if getattr...`) already handles that.